generators/JSONtoTests.jl

```diff
@@ -0,0 +1,138 @@
+# This code generates the runtets.jl file automatically for some lucky exercises that work with this script.
+
+# NOTE: This is a very experimental version of this idea. The code is not all that well written,
+# and may not work on most exercises. If it works, it might save you hours of time writing that exercise.
+# If it doesn't it will throw an error and quit.
+
+using JSON, HTTP
+
+# Convert an array to a array in string form
+# eg. 4-element Array{Int64,1}: 1 2 3 4 --> "[1, 2, 3, 4]"
+function array_to_str(array)
+    # Start array with opening bracket
+    str = ["["]
+    # If array is empty, return an empty array
+    if length(array) == 0
+        return "[]"
+    else
+    # Else, for each item in array, add it to array
+        for item in array
+            # If item is a string, enclose it in quotations and add it to our array
+            if typeof(item) == String
+                push!(str, "\"$item\", ")
+            # If item is an array, call function recursively to add it to our array
+            elseif typeof(item) == AbstractArray
+                push!(str, array_to_str(item))
+            # Else, if item is something else, just write it to array
+            else                
+                push!(str, "$item, ")
+            end
+        end
+
+        # Make it a string and remove the last extra comma and space from array
+        truncated_str = join(str)[1:end-2]
+
+        # Return final array string by closing with bracket
+        return string(truncated_str, "]")
+    end
+end
+
+function main(exercise_name)
+    # Get the exercise the user is working on from command line argument
+    current_exercise = exercise_name
+
+    # Download the test data for the exercise the user is working on and parse the JSON.
+    get_json = HTTP.get("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/$current_exercise/canonical-data.json")
```

Comment:
Not all exercises have canonical data. Perhaps a custom error message for those cases will look nicer than whatever the default error would look like.

Reply:
This might be causing some errors! Thanks for pointing this out.